src/libraries/System.Private.CoreLib/src/System/IO/Enumeration/FileSystemEntry.Unix.cs

```diff
@@ -11,6 +11,10 @@ namespace System.IO.Enumeration
     /// </summary>
     public unsafe ref partial struct FileSystemEntry
     {
+        // This value originated from NAME_MAX + 1 and was used as the size for DirectoryEntry.Name (d_name).
+        // It was later revealed that some filesystems can have filenames larger than that; however,
+        // it's still a reasonable size for a decoding buffer. If it's not enough, we fall back to allocating a string.
+        private const int DecodedNameBufferLength = 256;
```

Comment:
```suggestion
        // A reasonable size for a decoding buffer. If it's not enough, we fall back to allocating a string.
        // Some filesystem can have filenames larger than that.
        private const int DecodedNameBufferLength = 256;
```
We do not need to describe the whole history in the comment.